### ISSUE TITLE
Enhance custom symbol conditional statements to also work on characters/strings 

### ISSUE THREAD

**Issue by KristofKoch** [GenericMappingTools/gmt#316]
Conditional tests in a custom symbol are a great way to change to symbol depending on data passed to the symbol. This works already great for numerical values. 

I'd like to request to **make conditional testing available for characters/strings as well**. The _is equal_ and _is not equal_ operators would of interest. This will be useful when the symbol contains text or single characters. 

# Current situation:

A single character is passed to the symbol and the symbol has to display the character in a special way. Currently I have to pass the character plus a "control variable" where a conditional statement can act on to the symbol. 

## Placement file:
````
175.9   -38.8   1  A
173.75  -39.25  2  B
````

## Symbol code snippet:
````
if $1 == 1 then {
    0   0.1  0.8 $t l -W- -Gblack
} elseif $1 == 2 then {
    0  -0.1  0.8 $t l -W- -Gblack
}
````

# Improved Situation:

No need for the "control variable" when it is possible to test against characters/strings:

## Placement file:
````
175.9   -38.8   A
173.75  -39.25  B
````

## Symbol code snippet:
````
if $t == A then {
    0   0.1  0.8 $t l -W- -Gblack
} elseif $t == B then {
    0  -0.1  0.8 $t l -W- -Gblack
} ...
````

The placement file doesn't need redundant information and the symbol can act on the variable directly.

This feature request is inspired by [takaya nambas question in the forum about circled characters](http://gmt.soest.hawaii.edu/boards/1/topics/7982). The justification CM doesn't help much if you want to center a character in a circle. The optical middle of a character isn't always the geometric center. The letter "A" needs to sit a bit higher while the letter "W" needs to sit lower to appear visually pleasing centered. Another use case are warning levels like "dormant", "elevated", "active", "erupting" for volcanos …

**Comment by KristofKoch** [comment]
As stale bot got this issue in its claws – any news on this one?

**Comment by PaulWessel** [comment]
Hi @KristofKoch, I just tried it out and perhaps it already works.  I did this:
```
cat << EOF > t.txt
0	0	A
1	1	B
EOF

```
Then I made this custom symbol (text.def):
```
#
N: 1 s
if $t == A then {
    0   0.1  0.8 $t l -W- -Gred
} elseif $t == B then {
    0  -0.1  0.8 $t l -W- -Gblue
}

```

and ran

`gmt psxy -R-1/3/-1/3 -JX4i -P -Baf -Sktext/0.5i t.txt > t.ps
`

and I got a plot with the read A and blue B.  Give it a try and see what breaks.

**Comment by KristofKoch** [comment]
Hi @PaulWessel, a big thank you for implementing this! I’m currently on vacation. I will be able to do a simple test drive on Wednesday. A proper test has to wait another 2.5 weeks. 

A quick question without having tried it yet: Which operants are supported?

Are both

== IS EQUAL
!= IS NOT EQUAL

supported?

Thanks again,
Kristof

**Comment by PaulWessel** [comment]
No worries @KristofKoch.  I think all operants are supported (< <= = != etc) it is just lexical comparison so A is < B and != B, etc.  But my test only did the == for now.  Should work - it is just testing the output from strcmp which is negative, zero, or positive.

**Comment by KristofKoch** [comment]
Hi @PaulWessel, I tried hard breaking it and I did not succeed. Thank you again for implementing it. 

I noted one peculiarity which you might want to add to the documentation: The test is case-sensitive. I found myself scratching my head figuring out what was going on:

`$t > BA` does match `BB` but not `Bb` nor `bB`. 

In hindsight it is pretty clear but for some reason I assumed it to be case-agnostic. Maybe some future users will have the same thought as I did and a little note might be helpful to them. 

All the best,
Kristof
